core/sql/ustat/hs_la.cpp

```diff
@@ -994,13 +999,118 @@ Lng32 HSHiveTableDef::DescribeColumnNames()
 }
 
 //=====================================================
+//
+NAString HSHbaseTableDef::getHistLoc(formatType format) const
+{
+  if ( HSGlobalsClass::isNativeHbaseCat(getCatName(format))) {
+    return HBASE_STATS_CATALOG "." HBASE_STATS_SCHEMA;
+  } else {  
+    NAString name(getCatName(format));
+    name.append(".");
+    name.append(getSchemaName(format));
+    return name;
+  }
+}
+
+static 
+Lng32 CreateExternalTable(const NAString& catName, const NAString& schName, const NAString& nativeTableName)
+{
+   HSLogMan *LM = HSLogMan::Instance();
+   if (LM->LogNeeded())
+      {
+        snprintf(LM->msg, sizeof(LM->msg), "Creating external table table for %s on demand.",
+                          nativeTableName.data());
+        LM->Log(LM->msg);
+      }
+
+   // do not have to worry about the catalog and schema for the new external table 
+   // here. These names will be determined by the processing logic. 
+   NAString ddl = "CREATE EXTERNAL TABLE ";
+   ddl.append(nativeTableName);
+   ddl.append(" FOR ");
+   ddl.append(catName);
+   ddl.append(".");
+   ddl.append(schName);
+   ddl.append(".");
+   ddl.append(nativeTableName);
+   ddl.append(" ATTRIBUTE ");
+
+   HSTranMan *TM = HSTranMan::Instance();
+   if (TM->InTransaction())
+      ddl.append(" AUDIT");
+   else
+      ddl.append("NO AUDIT");
+
```

Comment:
I always assumed that external tables are non-audited.  You are creating them either audited or non-audited.  Since these tables do not contain any Trafodion data, is there any reason to make them audited?